eslint-plugin-prettier.js

```diff
@@ -401,48 +402,55 @@ module.exports = {
               usePrettierrc &&
               prettier.resolveConfig &&
               prettier.resolveConfig.sync
-                ? prettier.resolveConfig.sync(context.getFilename(), {
+                ? prettier.resolveConfig.sync(sourceFileName, {
                     editorconfig: true
                   })
                 : null;
             const prettierOptions = Object.assign(
               {},
               prettierRcOptions,
               eslintPrettierOptions,
-              { filepath: context.getFilename() }
+              { filepath: sourceFileName }
             );
-
             const prettierSource = prettier.format(source, prettierOptions);
-            if (source !== prettierSource) {
-              const differences = generateDifferences(source, prettierSource);
-
-              differences.forEach(difference => {
-                switch (difference.operation) {
-                  case OPERATION_INSERT:
-                    reportInsert(
-                      context,
-                      difference.offset,
-                      difference.insertText
-                    );
-                    break;
-                  case OPERATION_DELETE:
-                    reportDelete(
-                      context,
-                      difference.offset,
-                      difference.deleteText
-                    );
-                    break;
-                  case OPERATION_REPLACE:
-                    reportReplace(
-                      context,
-                      difference.offset,
-                      difference.deleteText,
-                      difference.insertText
-                    );
-                    break;
-                }
-              });
-            }
+
+            prettier.getFileInfo(sourceFileName, {
+              ignorePath: '.prettierignore'
+            }).then((sourceFileInfo) => {
```

Comment:
I don't think this will work -- ESLint rules are required to be synchronous. Does prettier provide a synchronous API for this?

Reply:
Yes! I tried both ways so I can update this to use the `.sync` version 👍